emscripten.py

```diff
@@ -1417,6 +1417,10 @@ def asmjs_mangle(name):
       # Don't include Invoke wrapper names (for asm.js-style exception handling)
       # in metadata[declares], the invoke wrappers will be generated in
       # this script later.
+      import_type = parts[3][1:]
+      if import_type == 'memory':
```

Comment:
Here we use these `'declares'` to determine what needs to be linked in via JS.
In `test_exceptions_std` and `test_bad_typeid`, we enable `ERROR_ON_UNDEFINED_SYMBOLS`, which causes  the test to fail on functions it can't find in one of the JS libraries. The JS libraries aren't responsible for linking in memory.
Normally undefined symbols are a warning, printed to stderr. Except on d8 where it's silently routed nowhere because there is no stderr :)

asm2wasm doesn't have this problem because it does the JS linking before even running asm2wasm, so memory hasn't entered the picture yet.

The fix here basically just says "ignore memories, try to link functions, assert that we aren't trying to import globals or tables just yet."